### PR TITLE
Initial Layout

### PR DESCRIPTION
**Pull request by ernestognw**
Project dashboard layout was added. 

Although I tried to follow the Javascript style guides from Google, I had some questions regarding the following:

1. The docs about import statements are not pretty clear since there are destructured imports, renamed imports, imports with renames, etc. Probably I'm ignoring something, but the criteria I followed was:

    - First, default imports alphabetically
    - Second, destructured imports (with destructured alphabetized)

    Hope to get some opinions about this.

2. The JSDoc requires me to write the `@return` and `@param` in every React component. This is a pattern that we'll see a lot, and I have some comments about it

    - `@return` is always typed the same. Since we're returning a React Node, I think it is pretty clear when we're writing a component, so I see no `@return` need. I just want to know what you think.
    - Every React component is required by ESLint to have their `PropTypes` which is the internal spec for the params received, so when writing `@param`'s, this info is duplicated. That's why, again, I think writing JSDoc here is unnecessary since React PropTypes are more standard.

### General updates
- **[Material UI](https://material-ui.com/) Setup**: This is the UI library we agreed to use
- **[Styled Components Setup](https://styled-components.com/)**: This is a css-in-js tool to write dynamic and custom components
- **[React Router](https://reactrouter.com/web/guides/quick-start) Setup**: Library to handle client-side routing
- **Layout added**: Basic navbar and routes in drawer were added
- **Basic routes**: About and mixed content routes were added.
- **Routes Tests**
- **Layout Tests**
- **App Tests**

**Note**: Please review and approve #1 first

## Screenshots
# Screenshot 1
# Screenshot 2

### CODE REVIEW

**Comment by JonathanChavezTamales** [comment]
Great! LGTM too

File: public/css/main.css [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/* Links style reset */
+body a {
+  display: contents;
```
Comment (livvielin) [inline public/css/main.css:3]: The declarations should be alphabetized.

Comment (ernestognw) [reply public/css/main.css]: Done

**Comment by ernestognw** [comment]
> Haven't finished reviewing yet, but would you be able to specify which things changed from https://github.com/googleinterns/step180-2020/pull/1/files (since that one isn't merged yet) and maybe provide a screenshot? These can help with reviewing. Thanks!

Np! I've solved the comments on #1 and updated this Pull Request, now you're able to only see the current changes.
Also, I added some screenshots in the description.

lmk if you need anything else.

File: src/views/about/index.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+
+/**
+ * About View
+ *
+ * This component should have everything to be rendered
+ * under /about route. It is intended to have a description
+ * about the data and what we're doing with the app
```
Comment (livvielin) [inline src/views/about/index.js:8]: nit: We generally try to avoid using "we" in comments. I would also recommend just saying what this component does rather than the intention. Maybe rephrase to something like "It contains a description of the data and the purpose of the app."

File: src/utils/test-utils.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    </StylesProvider>
+  );
+};
+
+AllTheProviders.propTypes = {
+  children: PropTypes.node.isRequired,
+};
+
+/**
+ * Custom render is an adaptatoion of original render
```
Comment (livvielin) [inline src/utils/test-utils.js:44]: nit: adaptation

File: src/utils/test-utils.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import PropTypes from 'prop-types';
+import {MemoryRouter} from 'react-router-dom';
+import {theme} from '../config/theme';
+import {ThemeProvider as StyledProvider} from 'styled-components';
+import {ThemeProvider, StylesProvider} from '@material-ui/core/styles';
+import React from 'react';
+import {render} from '@testing-library/react';
+
+/**
+ * This providers are used to setup context needed
```
Comment (livvielin) [inline src/utils/test-utils.js:18]: nit: These providers

File: src/utils/test-utils.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  );
+};
+
+AllTheProviders.propTypes = {
+  children: PropTypes.node.isRequired,
+};
+
+/**
+ * Custom render is an adaptatoion of original render
+ * with context providers to get the tests work.
```
Comment (livvielin) [inline src/utils/test-utils.js:45]: nit: tests **to** work

File: src/utils/test-utils.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
+/**
+ * @fileoverview Testing utils
+ *
+ * In this file, original test functions and utils are
+ * overridden to provide some customizations like a
+ * custom render with every needed wrapper to get
+ * the testings work
```
Comment (livvielin) [inline src/utils/test-utils.js:7]: nit: the tests to work

File: src/config/theme/index.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import {createMuiTheme} from '@material-ui/core/styles';
+
+const theme = createMuiTheme({
+  // Add special theming using this reference: https://material-ui.com/customization/default-theme/
+  palette: {
+    primary: {
+      light: '#63ccff',
```
Comment (livvielin) [inline src/config/theme/index.js:7]: nit: either put these in alphabetical order, or an intuitive order? I think main, light, dark, contrast makes sense to me, but feel free to not do this if it already makes sense in current order.

File: src/layouts/main/styles/functions.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+ * @param {object} theme
+ * @return {css} drawer styles when closed
+ */
+const getClosedDrawerStyles = (theme) => css`
+  transition: ${theme.transitions.create('width', {
+    easing: theme.transitions.easing.sharp,
+    duration: theme.transitions.duration.leavingScreen,
+  })};
+  overflow-x: hidden;
+  width: ${theme.spacing(9) + 1}px;
```
Comment (livvielin) [inline src/layouts/main/styles/functions.js:30]: Curious why these numbers were chosen and why adding 1?

File: src/layouts/main/styles/functions.js [hunk 9]
```diff
@@ -0,0 +1,74 @@
+import {css} from 'styled-components';
+import {drawerWidth} from './constants';
+
+/**
+ * To return drawer styles when is open.
```
Comment (livvielin) [inline src/layouts/main/styles/functions.js:5]: nit: For comments, usually we like to start with a verb. e.g. "Returns drawer styles..."

For here and in other places.

File: src/layouts/main/styles/functions.js [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+`;
+
+/**
+ * To return drawer styles according to
+ * the opened state
+ *
+ * @param {{open: boolean, theme: object}} props
+ * @return {css} styles according to open state
+ */
+const drawerToggles = ({open, theme}) => {
```
Comment (livvielin) [inline src/layouts/main/styles/functions.js:44]: Maybe name this toggleDrawer instead? Tend to like the verb since we're doing something with this function.

File: src/layouts/main/styles/functions.js [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+`;
+
+/**
+ * To return active styles for navigation tab
+ * only when is active
+ *
+ * @param {{active: boolean, theme: object}} props
+ * @return {css} styles according to ac state
+ */
+const tabToggle = ({active, theme}) => {
```
Comment (livvielin) [inline src/layouts/main/styles/functions.js:70]: I'm not sure I understand the function naming for this? Also, would it be possible to merge this function with getActiveTabStyles?

File: src/layouts/main/styles/functions.js (hunk 9 above)
Comment (ernestognw) [reply src/layouts/main/styles/functions.js]: Done, also I rewrote almost every comment to try to fit into this recommendation.

File: src/layouts/main/styles/functions.js (hunk 8 above)
Comment (ernestognw) [reply src/layouts/main/styles/functions.js]: tbh, I copied some style examples from the official docs, and adapted accordingly, I removed the + 1 since it actually has no purpose.

Regarding the spacing, the spec is [here](https://material-ui.com/customization/spacing/), but there is no real motivation to select `theme.spacing(9)`, just an opinionated decision from my side.

Let me know if you have a clue on how we can organize these decisions better.

File: src/layouts/main/styles/functions.js (hunk 10 above)
Comment (ernestognw) [reply src/layouts/main/styles/functions.js]: I decided to name it `getToggleDrawerStyles`. I think `toggleDrawer` sounds more like a function to handle component functionality. I just don't want to create confusion with the purpose of these functions.

Let me know if it sounds better, I was thinking about it for a few minutes

File: src/layouts/main/styles/functions.js (hunk 11 above)
Comment (ernestognw) [reply src/layouts/main/styles/functions.js]: Some context: I had some problems with the css-in-js, since it uses indentation inside JS template literals, and the linter shows an indentation error. That's why I decided to put these styles in subfunctions since it respects the indentation. I'm attaching an SS for that:

ESLint Error
Original function

That been said, I tried to use only `getActiveTabStyles` as suggested, and it worked, so I replaced it 😄 , I'm just saying this just to show that we might have to add a special rule in the ESLint configuration to avoid indentation errors. In that case, I can simplify the way we use these functions, which is important since it'll be frequent to have them.

File: src/utils/test-utils.js (hunk 6 above)
Comment (ernestognw) [reply src/utils/test-utils.js]: Done!

File: src/utils/test-utils.js (hunk 4 above)
Comment (ernestognw) [reply src/utils/test-utils.js]: Done!

File: src/utils/test-utils.js (hunk 3 above)
Comment (ernestognw) [reply src/utils/test-utils.js]: Done!

File: src/utils/test-utils.js (hunk 5 above)
Comment (ernestognw) [reply src/utils/test-utils.js]: Done, also corrected the same problem at the top of the document

File: src/views/about/index.js (hunk 2 above)
Comment (ernestognw) [reply src/views/about/index.js]: Awesome, I readapted almost every comment to avoid "we" and start with a verb

File: .gitignore [hunk 12]
```diff
@@ -24,4 +24,8 @@ yarn-debug.log*
 yarn-error.log*
 
 # credentials
+gcloud_creds.json
```
Comment (ernestognw) [inline .gitignore:27]: Added this in advance to avoid future accidents, since @JonathanChavezTamales and @SofiaVega are using this file to handle credentials.

I noticed some branches already have this, but as early as we get this to master branch, the better.

Comment (livvielin) [reply .gitignore:27]: lgtm

File: src/layouts/main/styles/functions.js [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+`;
+
+/**
+ * Returns drawer styles according to
+ * the opened state
+ *
+ * @param {{open: boolean, theme: object}} props
+ * @return {css} styles according to open state
+ */
+const getToggleDrawerStyles = ({open, theme}) => {
```
Comment (livvielin) [inline src/layouts/main/styles/functions.js:44]: nit: maybe just getDrawerStyles? Since you're not actually toggling

File: src/utils/test-utils.js [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+/**
+ * Setup context needed in every render, so test could pass
+ * and also replicate a pretty similiar environment to the
+ * one when app is running
+ *
+ * @param {{children: ReactNode}} params
+ *    children is the component to be rendered in tests
+ * @return {ReactNode} Enhanced children with providers
+ */
+const AllTheProviders = ({children}) => {
```
Comment (livvielin) [inline src/utils/test-utils.js:25]: nit: maybe just AllProviders, since "The" doesn't seem to be necessary?

Comment (ernestognw) [reply src/utils/test-utils.js]: Done!

File: src/layouts/main/styles/functions.js (hunk 13 above)
Comment (ernestognw) [reply src/layouts/main/styles/functions.js]: Done!
